runtime/oti/j9.h

```diff
@@ -356,6 +356,7 @@ static const struct { \
 #define J9_IS_FIELD_FLATTENED(fieldClazz, romFieldShape) FALSE
 #define J9_IS_NULL_RESTRICTED_FIELD_FLATTENED(fieldClazz, romFieldShape) FALSE
 #define J9_VALUETYPE_FLATTENED_SIZE(clazz)((UDATA) 0) /* It is not possible for this macro to be used since we always check J9_IS_J9CLASS_FLATTENED before ever using it. */
+#define J9CLASS_GET_NULLRESTRICTED_ARRAY(clazz) NULL
```

Comment:
This is part of the real change since `J9CLASS_GET_NULLRESTRICTED_ARRAY` is only defined in the VM https://github.com/eclipse-openj9/openj9/pull/20132 when `J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES` is defined, but the JIT code needs to build for both Valhalla and non-Valhalla build. 

@theresa-m Sorry to bring it up late. Would it still be possible to add this line of VM change in `runtime/oti/j9.h` into https://github.com/eclipse-openj9/openj9/pull/20132?

```
--- a/runtime/oti/j9.h
+++ b/runtime/oti/j9.h
@@ -356,6 +356,7 @@ static const struct { \
 #define J9_IS_FIELD_FLATTENED(fieldClazz, romFieldShape) FALSE
 #define J9_IS_NULL_RESTRICTED_FIELD_FLATTENED(fieldClazz, romFieldShape) FALSE
 #define J9_VALUETYPE_FLATTENED_SIZE(clazz)((UDATA) 0) /* It is not possible for this macro to be used since we always check J9_IS_J9CLASS_FLATTENED before ever using it. */
+#define J9CLASS_GET_NULLRESTRICTED_ARRAY(clazz) NULL
 #endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
```

Reply:
I'm going to say yes... but I can add it in the follow up that will create an `arrayClass` before every `nullRestrictedArrayClass` that was discussed in https://github.com/eclipse-openj9/openj9/issues/19914#issuecomment-2311388833

Reply:
That works too! Thank you very much @theresa-m! I'll remove this change from this PR once the new PR opens